electrum/lnpeer.py

```diff
@@ -1885,17 +1924,98 @@ def verify_signature(tx, sig):
+                            # MUST fail the channel
+                            self.lnworker.schedule_force_closing(chan.channel_id)
+                            raise Exception("Their fee is not in the overlap region, we force closed.")
+                        # otherwise:
+                        else:
+                            our_fee = their_fee
+                            # MUST reply with the same fee_satoshis.
+                            send_closing_signed()  # peer publishes
+                            break
+                    # otherwise (it is not the funder):
+                    else:
+                        # if it has already sent a closing_signed:
+                        if fee_range_sent:
+                            # if fee_satoshis is not the same as the value it sent:
+                            if their_fee != our_fee:
+                                # MUST fail the channel
+                                self.lnworker.schedule_force_closing(chan.channel_id)
+                                raise Exception("Expected the same fee as ours, we force closed.")
+                        # otherwise:
+                        else:
+                            # MUST propose a fee_satoshis in the overlap between received and (about-to-be) sent fee_range.
+                            our_fee = (overlap_min + overlap_max) // 2
+                            send_closing_signed()
+                            continue
+            # otherwise, if fee_satoshis is not strictly between its last-sent fee_satoshis
+            # and its previously-received fee_satoshis, UNLESS it has since reconnected:
+            elif their_previous_fee and not (min(our_fee, their_previous_fee) < their_fee < max(our_fee, their_previous_fee)):
+                # SHOULD fail the connection.
+                raise Exception('Their fee is not between our last sent and their last sent fee.')
+            # otherwise, if the receiver agrees with the fee:
+            elif abs(their_fee - our_fee) <= 1:  # we cannot have another strictly in-between value
+                # SHOULD reply with a closing_signed with the same fee_satoshis value.
                 our_fee = their_fee
+                send_closing_signed()  # peer publishes
                 break
-            # this will be "strictly between" (as in BOLT2) previous values because of the above
-            our_fee = (our_fee + their_fee) // 2
-            # another round
-            send_closing_signed()
-        # the non-funder replies
-        if not chan.constraints.is_initiator:
+            # otherwise:
+            else:
+                # MUST propose a value "strictly between" the received fee_satoshis and its previously-sent fee_satoshis.
+                our_fee_proposed = (our_fee + their_fee) // 2
+                if not (min(our_fee, their_fee) < our_fee_proposed < max(our_fee, their_fee)):
+                    our_fee_proposed += (their_fee - our_fee) // 2
```

Comment:
I do not understand what this code is doing.
First, I suppose that the intent is to set `our_fee`, not `our_fee_proposed`. 
So, maybe you wanted to write something like `our_fee = our_fee_proposed + (their_fee - our_fee)//2`
Second, if `abs(our_fee - their_fee) > 1`, then `(our_fee + their_fee) // 2` will always be strictly between the values, as explained in the comment on line 1892. Therefore, that code seems unnecessary.

Reply:
yes, you're right, I think I experienced an off-by-one somewhere here and tried to fix it that way